src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -180,4 +184,19 @@ public void updatePassword(@PathVariable Long id, @RequestBody User user) {
         userRepository.save(dbUser);
     }
 
+
+    /**
+     * Get's the current logged in {@link User}.
+     *
+     * @return the {@link User}
+     */
+    @GetMapping("/me")
+    @ResponseBody
+    public User getLoggedInUser(HttpServletRequest request) {
+        String username = request.getUserPrincipal().getName();
+        User user = userRepository.findByUsername(username);
+        if (user == null)
+            throw new EntityNotFoundException(User.class);
+        return user;
```

Comment:
The best to do here is to change the repository so it can return an `Optional<User>` instead of a `User` and here you can do just this:
```
  return userRepository.findByUsername(username)
    .orElseThrow(()-> new EntityNotFoundException(User.class));
```